company-lsp.el

```diff
@@ -131,10 +131,11 @@ P and MSG are the parameters from `lsp--parser-on-message'."
 	 (json-data (json-read-from-string msg)))
     (pcase (lsp--get-message-type json-data)
       ('response
-       (when-let* ((id-msg (gethash "id" json-data nil))
-		   (callback (gethash id-msg company-lsp--pending-requests)))
-	 (remhash id-msg company-lsp--pending-requests)
-	 (company-lsp--on-completion (lsp--parser-response-result p) callback))))))
+       (let* ((id-msg (gethash "id" json-data nil))
```

Comment:
let* may throw error if id-msg is null

Can you use -when-let* and add dash as the dependency?

Reply:
What? I don't understand ^^ I tested with messages that does not contain an id, so id-msg was nil and no error was thrown.

Reply:
My bad. Missed the (when (and id-msg callback) line

It's good.